### PR TITLE
Redux additions; and action and reducer additions. 

### PR DESCRIPTION
**Pull request by CrystallineButterfly**
Hi there, this is a PR, 

To show use cases of redux and action and reducers. 

All the best-K

### CODE REVIEW

**Review by topeogunleye — COMMENTED** [review]
Hi @CrystallineButterfly, 

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request ⚠️

   - You have submitted a project that has failing linters, Kindly make sure your linters are all passing before submitting the PR.

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag @topeogunleye in your question so I can receive the notification.


**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

>Invalid Code Review Request **does not count** into the code reviews limit.

**Comment by CrystallineButterfly** [comment]
Hi there @topeogunleye Thank you for the review, one issue I am having is that this linter error doesn't exist in my VS code, and only on GitHub, do you have a fix for this pls? 

All the best-K

File: src/App.js [hunk 1]
```diff
@@ -1,25 +1,49 @@
...
+//           <Route exact path="/books" element={<Books />} />
+//           <Route path="/catergories" element={<Catergories />} />
+//         </Routes>
+//       </div>
+//     </Provider>
+//   );
+// }
+
+// export default BookStore;
+
```
Comment (rotimiazeez) [inline src/App.js]: - Kindly remove all commented code to main the readability of your code 👍🏽

File: src/components/books/AddBooks.js [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+// //       Title
+// //       <input type="text" property="title" />
+// //       Author
+// //       <input type="text" property="author" />
+// //       <button type="submit">Add Book</button>
+// //     </div>
+// //   );
+// // }
+
+// // export default AddBooks;
```
Comment (rotimiazeez) [inline src/components/books/AddBooks.js]: - Kindly remove all commented code to main the readability of your code 👍🏽

**Review by rotimiazeez — CHANGES_REQUESTED** [review]
# Hi @CrystallineButterfly  👏🏽, 
## Good job so far! 👍🏽 
### Summary ♨️ 
- No Linter error ✔️ 
- Followed git-flow 👏🏽 
- Good and descriptive PR 💯 
There are some issues that you still need to work on to go to the next project but you are almost there!
<FEEL FREE TO CHANGE THIS MESSAGE TO YOUR PERSONAL WORDS>

## Highlights


## Required Changes ♻️
_Check the comments under the review._
- [ ] Please note that when I start the app locally I get an empty page, kindly look into it. 😉  

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/components/books/AddBooks.js (hunk 2 above)
Comment (CrystallineButterfly) [reply src/components/books/AddBooks.js]: Thank you for the review! @rotimiazeez I have implemented the changes now!

All the best-K

**Review by Deepakdanger — CHANGES_REQUESTED** [review]
Hi @CrystallineButterfly, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.

 ## Required Changes ♻️

- [x] Please check the first reviewer comment that when I start the app locally I get an empty page, still, this is an issue. kindly look into it. :wink:

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

- N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/redux/configureStore.js [hunk 3]
```diff
@@ -0,0 +1,16 @@
+/* eslint-disable import/no-extraneous-dependencies */
+/* eslint-disable no-underscore-dangle */
+import { configureStore } from '@reduxjs/toolkit';
```
Comment (thecodechaser) [inline src/redux/configureStore.js:3]: - [x] As requested by the previous reviewer when I start the app locally I get an empty page and this error. Kindly look into it and fix it.

image

**Review by thecodechaser — CHANGES_REQUESTED** [review]
## STATUS: CHANGES REQUIRED ♻️ 

## Hi @CrystallineButterfly👋, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.

## To Highlight 🎉
- No linter errors. :heavy_check_mark:
- Gitflow is followed. :heavy_check_mark:
- Professional README file. :heavy_check_mark:
- PR has a good title and summary. ✔️



 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by topeogunleye** [comment]
> Hi there @topeogunleye Thank you for the review, one issue I am having is that this linter error doesn't exist in my VS code, and only on GitHub, do you have a fix for this pls?
> 
> All the best-K

Hi @CrystallineButterfly , Sorry for my late reply. I'm glad you found a solution.

File: src/redux/configureStore.js (hunk 3 above)
Comment (CrystallineButterfly) [reply src/redux/configureStore.js:3]: Ok thank you @thecodechaser Hope you are blessed and well!

File: src/App.jsx [hunk 4]
```diff
@@ -0,0 +1,27 @@
+import {
+  BrowserRouter as Router,
+  Routes,
+  Route,
+} from 'react-router-dom';
+import { Provider } from 'react-redux';
+import Book from './Components/Book';
+import store from './Redux/configureStore';
```
Comment (youmari) [inline src/App.jsx]: - [x] the problem is still there that the previous reviewer stated to solve ti please try to change `Redux` to `redux` in order the error disappear

**Review by youmari — APPROVED** [review]
# Project Approved :trophy: :tada: 🟢 

Hi @CrystallineButterfly 

Your project is complete!  You have done it exceptionally well. it's time to merge it :100: 
Congratulations! 🎉 

### To Highlight :+1: 

:heavy_check_mark: No linters error.
:heavy_check_mark: Details PR title and description.
:heavy_check_mark: Good commit messages.

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification. You can also connect with me on [slack](https://microverse-students.slack.com/team/U02FTTCCU2V)

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Pull request merged**